### PR TITLE
ImplementsComment likes

### PR DESCRIPTION
**Pull request by janko101**
## Changes proposed in this Pull Request:
- Adds CommentLike model
- Adds CommentLikeController 
- Add a like to a specific comment
- remove an added like for a specific comment
- get all likes for a specific comment

PBI-47
`As a pet owner, I want to like a comment, so that I can express my reaction to it`

PBI-48
`As a pet owner, I want to unlike a comment, so that I can express my reaction to it`

### CODE REVIEW

**Review by deciocandido — APPROVED** [review]
Great work!!

File: src/main/java/se/kth/sda/skeleton/commentlikes/CommentLikeController.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    public ResponseEntity<List<CommentLike>> getAllLikes(@PathVariable Long commentId){
+        return ResponseEntity.ok(commentLikeService.getAllLikes(commentId));
+    }
+
+    /**
+     * Handler method for endpoint "/posts/{postId}/likes" with DELETE HttpRequest. Invokes a HTTP {@link ResponseStatus} of
+     * NO_CONTENT. Deletes the like made by the logged in user for post given by {@code postId}
+     * @param postId the post for which the user's like will be removed
+     */
+    @DeleteMapping("/{postId}/likes")
```
Comment (eeels22) [inline src/main/java/se/kth/sda/skeleton/commentlikes/CommentLikeController.java]: `("/{commentId}/likes")`

File: src/main/java/se/kth/sda/skeleton/commentlikes/CommentLikeController.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    @GetMapping("/{commentId}/likes")
+    public ResponseEntity<List<CommentLike>> getAllLikes(@PathVariable Long commentId){
+        return ResponseEntity.ok(commentLikeService.getAllLikes(commentId));
+    }
+
+    /**
+     * Handler method for endpoint "/posts/{postId}/likes" with DELETE HttpRequest. Invokes a HTTP {@link ResponseStatus} of
+     * NO_CONTENT. Deletes the like made by the logged in user for post given by {@code postId}
+     * @param postId the post for which the user's like will be removed
+     */
```
Comment (eeels22) [inline src/main/java/se/kth/sda/skeleton/commentlikes/CommentLikeController.java]: amend to reflect comment

File: src/main/java/se/kth/sda/skeleton/commentlikes/CommentLikeRepository.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package se.kth.sda.skeleton.commentlikes;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+/**
+ * Represents the database of domain type Comment as a JPA Repository
```
Comment (eeels22) [inline src/main/java/se/kth/sda/skeleton/commentlikes/CommentLikeRepository.java]: `Represents the database of domain type **CommentLike** as a JPA Repository`

File: src/main/java/se/kth/sda/skeleton/commentlikes/CommentLikeService.java [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+import se.kth.sda.skeleton.exception.ForbiddenException;
+import se.kth.sda.skeleton.exception.ResourceNotFoundException;
+import se.kth.sda.skeleton.postlikes.PostLike;
+import se.kth.sda.skeleton.posts.Post;
+import se.kth.sda.skeleton.user.User;
+import se.kth.sda.skeleton.user.UserRepository;
+import java.util.List;
+
+/**
+ * Represents the service layer. It contains defined functionalities according to business logic for PostLike.
```
Comment (eeels22) [inline src/main/java/se/kth/sda/skeleton/commentlikes/CommentLikeService.java]: ...business logic for `CommentLike`.

File: src/main/java/se/kth/sda/skeleton/commentlikes/CommentLikeController.java (hunk 1 above)
Comment (janko101) [reply src/main/java/se/kth/sda/skeleton/commentlikes/CommentLikeController.java]: Done

File: src/main/java/se/kth/sda/skeleton/commentlikes/CommentLikeController.java (hunk 2 above)
Comment (janko101) [reply src/main/java/se/kth/sda/skeleton/commentlikes/CommentLikeController.java]: Amended

File: src/main/java/se/kth/sda/skeleton/commentlikes/CommentLikeRepository.java (hunk 3 above)
Comment (janko101) [reply src/main/java/se/kth/sda/skeleton/commentlikes/CommentLikeRepository.java]: Done

File: src/main/java/se/kth/sda/skeleton/commentlikes/CommentLikeService.java (hunk 4 above)
Comment (janko101) [reply src/main/java/se/kth/sda/skeleton/commentlikes/CommentLikeService.java]: Done

**Review by eeels22 — APPROVED** [review]
Super work Janko

**Pull request merged**